compiler/x/codegen/DataSnippet.cpp

```diff
@@ -48,19 +49,29 @@ void
 TR::IA32DataSnippet::addMetaDataForCodeAddress(uint8_t *cursor)
    {
    // add dummy class unload/redefinition assumption.
+   bool needRelocation = cg()->comp()->compileRelocatableCode() || TR::Compiler->cls.classUnloadAssumptionNeedsRelocation();
```

Comment:
Can we simply use one query here? Can compilers doing AOT also return true for `classUnloadAssumptionNeedsRelocation()`?